### PR TITLE
Remove securesystemslib.util.TempFile

### PR DESCRIPTION
**Pull request by joshuagl**
**Fixes issue #**: #180 

**Description of the changes being introduced by the pull request**: remove the `securesystemslib.util.TempFile` class.

This was originally added to provide convenient handling of compressed and uncompressed file like objects however, those features of `securesystemslib.util.TempFile` are no longer used by tuf nor in-toto.
Furthermore `securesystemslib.util.TempFile` has some behavioural inconsistencies with a standard Python file object, which cause confusion for users of the class.

The following code changes are introduced:

1.  add a new helper `securesystemslib.util.persist_temp_file()` to create a copy of the temporary file object (first arg) at the path specified in the second argument in 4a36f97
2. remove all internal users of `securesystemslib.util.TempFile` in `securesystemslib.interface` in e594e66
3. remove the class itself and the associated unit tests in a6dca67

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: securesystemslib/util.py [hunk 1]
```diff
@@ -395,6 +95,34 @@ def get_file_details(filepath, hash_algorithms=['sha256']):
   return file_length, file_hashes
 
 
+def persist_temp_file(temp_file, persist_path):
+  """
+  <Purpose>
+    Copies 'temp_file' (a file like object) to a newly created non-temp file at
+    'persist_path' and closes 'temp_file' so that it is removed.
+
+  <Arguments>
```
Comment (lukpueh) [inline securesystemslib/util.py]: Would you mind adding `temp_file` to the argument section as well?

Comment (lukpueh) [reply securesystemslib/util.py]: Just found out about GitHub suggestions. :) Does this work for you?

```suggestion
  <Arguments>
  temp_file:
    A file-like object, e.g. as returned by 'tempfile.TemporaryFile()'.
     
```

Comment (joshuagl) [reply securesystemslib/util.py]: Great catch. I've added this to the first commit and force pushed an updated series.

Comment (lukpueh) [reply securesystemslib/util.py]: Hooray!

Comment (joshuagl) [reply securesystemslib/util.py]: > Just found out about GitHub suggestions. :) Does this work for you?

Apologies, I missed that. This change would have been fine.

Comment (lukpueh) [reply securesystemslib/util.py]: No worries, yours is actually better. :)

**Comment by joshuagl** [comment]
Excellent news, can't wait to see the releases. Do let me know if there's any way I can help. Otherwise I'll dive into some more tuf and ssl issues next week. 😊

**Pull request merged**